L3 Mathematical modelling of an electric vehicle /Calculating Energy Consumption {Software simulation in Scilab and Xcos} - Part 1.md

```diff
@@ -0,0 +1,200 @@
+
+
+## 3. Acquiring the WLTC Test cycle Data
+
+Click on the link below to Download the data, Make sure you store it in  the *Electric vehicles* folder created in the previous step.
+
+[Download WLTP-DHC-12-07e.xls](https://demo.pflms.com/markdown_attachments/1780/BBDYix0qAUzkv4FVzSxZ-w)
+
+## 4. Importing xls (Excel) data into Scilab and Xcos.
+
+There is a predefined Scilab function which can read the content of *.xls files. The Scilab function **xls_read()** reads a sheet from an excel file and saves the data in the Scilab workspace. 
+
+The **xls_read()** function reads an excel sheet given a logical unit on an excel stream and the position of the beginning of the sheet within this stream. It returns the numerical data and the strings contained by the excel cells.
+
+The **read_xls()** function can be used to read all sheets from an excel file in one function with a single function call.
+
+The function can be called as:
+```
+[Value,TextInd] = xls_read(fd,Sheetpos) 
+```
+**where:**
+
+- **fd** – a number: the logical unit on the Excel stream returned by xls_open() Scilab function
+- **Sheetpos** – a number: the position of the beginning of the sheet in the Excel stream. This position is one of those returned by xls_open()
+- **Value** – a matrix of numbers: the numerical data found in the sheet. The cells without numerical data are represented by NaN values
+- **TextInd** – a matrix of indices with the same size as Value. The 0 indices indicates that no string exists in the corresponding Excel cell. A positive index i points to the string SST(i) where SST is given by xls_open()
+
+**Don't worry, you don't have to type your own code. It will be given to you in the coming steps. This is just to make you understand what the code does**
+
+>Important note: Only excel file version (2003) are handled.
+
+### Let’s import the data describing the WLTP speed profile. The file can be found here[]:
+
+The data will be first imported into Scilab, stored into a variable, and used later in Xcos for simulation purposes.
+
+**Step 1**. Save the *.xls file after downloading, in the current Scilab working folder, i.e "Electric vehicles"
+
+**Step 2**. Open the *.xls file and examine the data you want to import. 
+
+
+![xls.jpg](https://demo.pflms.com/markdown_attachments/1781/qbn5dUHX0CuM6eMuLneloA)
+
+In this example, the data is located in the second sheet, named **WLTC_class_3**. We will import the WLTP speed profile, which consists of time values and speed values. 
+
+- **The time values**, in s, begin in the 8th row and 3rd column (C), with value 0. 
+
+- **The speed values**, in kph, begin in the 8th row and 5th column (E), with value 0.0. If you scroll down towards the end of the table, you’ll see that the last data points are in the row 1808.
+
+
+**Step 3**. Open SciNotes (Applications>Scinotes) and create a script file with the following content:
+
+```
+clear()
+clc()
+//Decode ole file, extract and open Excel stream
```

Comment:
This is the first mention of `ole` file in this lesson. We need to add a line about this or add an external link to Scilab website where this is discussed.